text/0000-feature-pruning.md

```diff
@@ -0,0 +1,224 @@
+- Start Date: 2016-12-07
+- RFC PR: (leave this empty)
+- Yarn Issue: (leave this empty)
+
+# Summary
+
+> Allow organizing dependencies in custom named groups
+
+At different points in a package's lifecycle, the package may require a
+different set of dependencies, currently represented as `devDependencies`,
+`peerDependencies`, `dependencies` and `optionalDependencies`.  While these sets
+of dependencies are expressive for the majority of existing node packages
+published on the public registry, some packages, and many applications, have
+lifecycles that are much more complicated than than the develop and deploy
+lifecycle that currently exists.  This rfc allows for packages to define custom
+dependency closures to match their lifecycle.
+
+# Motivation
+
+Better support for continous integration pipelines. Currently, there is a class

```

Comment:
Amusingly, I made this mistake just today -- I used [np](https://github.com/sindresorhus/np) to publish 0.0.2 of [react-link-to-inbox](https://www.npmjs.com/package/react-link-to-inbox) and listed [styled-components](https://github.com/styled-components/styled-components) as a devDependency instead of a regular dependency :joy: